### PR TITLE
Add ex_webrtc.exs

### PR DESCRIPTION
**Pull request by mickel8**

### CODE REVIEW

File: ex_webrtc.exs [hunk 1]
```diff
@@ -0,0 +1,258 @@
+Mix.install([
+  {:bandit, "~> 1.2"},
+  {:plug, "~> 1.15"},
+  {:websock_adapter, "~> 0.5.0"},
+  {:jason, "~> 1.4.0"},
```
Comment (wojtekmach) [inline ex_webrtc.exs]:
```suggestion
  {:jason, "~> 1.4"},
```

File: ex_webrtc.exs [hunk 2]
```diff
@@ -0,0 +1,258 @@
...
+  {:ex_webrtc, "~> 0.4.1"}
+])
+
+require Logger
+Logger.configure(level: :info)
```
Comment (wojtekmach) [inline ex_webrtc.exs]:
```suggestion
Logger.configure(level: :info)

defmodule Main do
  require Logger

  def main do
    ip = {127, 0, 0, 1}
    port = 8829

    {:ok, _pid} = Bandit.start_link(plug: Router, ip: ip, port: port)
    Logger.info("Visit http://#{:inet.ntoa(ip)}:#{port}")

    # unless running from IEx, sleep idenfinitely so we can serve requests
    unless IEx.started?() do
      Process.sleep(:infinity)
    end
  end
end
```

File: ex_webrtc.exs [hunk 3]
```diff
@@ -0,0 +1,258 @@
...
+port = 8829
+
+{:ok, _pid} = Bandit.start_link(plug: Router, ip: ip, port: port)
+
+Logger.info("Visit http://#{:inet.ntoa(ip)}:#{port}")
+
+# unless running from IEx, sleep idenfinitely so we can serve requests
+unless IEx.started?() do
+  Process.sleep(:infinity)
+end
```
Comment (wojtekmach) [inline ex_webrtc.exs]:
```suggestion
```

File: ex_webrtc.exs [hunk 4]
```diff
@@ -0,0 +1,258 @@
...
+    :ok = PeerConnection.set_remote_description(state.peer_connection, offer)
+
+    {:ok, answer} = PeerConnection.create_answer(state.peer_connection)
+    :ok = PeerConnection.set_local_description(state.peer_connection, answer)
+
+    answer_json = SessionDescription.to_json(answer)
+
+    msg =
+      %{"type" => "answer", "data" => answer_json}
+      |> Jason.encode!()
```
Comment (wojtekmach) [inline ex_webrtc.exs]:
```suggestion
    answer_json = SessionDescription.to_json(answer)
    msg = Jason.encode!(%{"type" => "answer", "data" => answer_json})
```

btw, JSON inside JSON huh? 😅

File: ex_webrtc.exs [hunk 5]
```diff
@@ -0,0 +1,258 @@
...
+    :ok = PeerConnection.add_ice_candidate(state.peer_connection, candidate)
+    {:ok, state}
+  end
+
+  defp handle_webrtc_msg({:ice_candidate, candidate}, state) do
+    candidate_json = ICECandidate.to_json(candidate)
+
+    msg =
+      %{"type" => "ice", "data" => candidate_json}
+      |> Jason.encode!()
```
Comment (wojtekmach) [inline ex_webrtc.exs]:
```suggestion
    candidate_json = ICECandidate.to_json(candidate)
    msg = Jason.encode!(%{"type" => "ice", "data" => candidate_json})
```

File: ex_webrtc.exs [hunk 6]
```diff
@@ -0,0 +1,258 @@
...
+
+  defp handle_webrtc_msg({:rtcp, packets}, state) do
+    for packet <- packets do
+      case packet do
+        {_track_id, %ExRTCP.Packet.PayloadFeedback.PLI{}} when state.in_video_track_id != nil ->
+          Logger.info("Received keyframe request. Sending PLI.")
+          :ok = PeerConnection.send_pli(state.peer_connection, state.in_video_track_id, "h")
+
+        _other ->
+          # do something with other RTCP packets
```
Comment (wojtekmach) [inline ex_webrtc.exs]:
```suggestion
```

**Review by wojtekmach — COMMENTED** [review]
This amazing, thanks for this script and all the underlying work!

**Comment by wojtekmach** [comment]
Thanks!

File: ex_webrtc.exs (hunk 4 above)
Comment (mickel8) [reply ex_webrtc.exs]: WebRTC  world :smiley:

**Comment by mickel8** [comment]
Thanks for improvements!

**Pull request merged**